### PR TITLE
Replace elastic.co with opensearch.org in opensearch.yml

### LINKED ISSUE #609: [CLEANUP] Replace elastic.co links from opensearch.yml
Replace `https://www.elastic.co/guide/en/elasticsearch/reference/index.html`  with opensearch doc link. Since we don't have the documentation published yet we will point it to `opensearch.org` for now. 

Issue to update doc link once we documentation online: #610

### PR DESCRIPTION
**Pull request by abbashus**
Signed-off-by: Abbas Hussain <abbas_10690@yahoo.com>

### Description
Cleans up `distribution/src/config/opensearch.yml` by replacing _elastic.co_ to _opensearch.org_
 
### Issues Resolved
#609 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by saratvemulapalli — APPROVED** [review]

File: distribution/src/config/opensearch.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@
 # the most important settings you may want to configure for a production cluster.
 #
 # Please consult the documentation for further information on configuration options:
-# https://www.elastic.co/guide/en/elasticsearch/reference/index.html
+# https://www.opensearch.org
```
Comment (saratvemulapalli) [inline distribution/src/config/opensearch.yml:11]: Probably add a TODO to update the URL when Docs link is available ?

Comment (harold-wang) [reply distribution/src/config/opensearch.yml:11]: > Probably add a TODO to update the URL when Docs link is available ?

Tianli or Himanshu might have know what is corresponding think that OpenSearch is using?

**Review by harold-wang — APPROVED** [review]

File: distribution/src/config/opensearch.yml (hunk 1 above)
Comment (adnapibar) [reply distribution/src/config/opensearch.yml:11]: What's the reasoning behind this change? Please see my comment on issue #609

**Comment by nknize** [comment]
start gradle check

**Pull request merged**